### Analisis exploratorio
Importo las librerias que voy a necesitar y cargo los dataset con los que voy a trabajar. Se trata de 3 dataset de tres áreas de una fábrica de hojas de papel. los datos se toman cada un minuto. Los dataset vienen tal como estan en la base de datos. Es decir sin limpiar.

Operaciones: Se encarga de la producción.\
Aditivos: Se encarga de la aplicación de productos químicos necesarios para la producción.\
Caliedad: Se encarga de controlar las variables de calidad importantes para los clientes.

In [442]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [352]:
df_operaciones = pd.read_csv('operaciones.csv')
df_aditivos = pd.read_csv('aditivos.csv')
df_calidad = pd.read_csv('calidad.csv')

C:\Users\sas\AppData\Local\Temp\ipykernel_27412\285350360.py:1: DtypeWarning: Columns (1,3,4,5,6,7,8,9,10,11,12,13) have mixed types. Specify dtype option on import or set low_memory=False.
  df_operaciones = pd.read_csv('operaciones.csv')
C:\Users\sas\AppData\Local\Temp\ipykernel_27412\285350360.py:2: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aditivos = pd.read_csv('aditivos.csv')
C:\Users\sas\AppData\Local\Temp\ipykernel_27412\285350360.py:3: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,14,15,16,17) have mixed types. Specify dtype option on import or set low_memory=False.
  df_calidad = pd.read_csv('calidad.csv')


### Explorando los dataset
los datos estan como Dtype object por lo que tengo que convertirlos a su tipo de dato para explorarlos mejor.\
-Primero convierto la columna 'Date' de cada dataset a formato datetime. Dado que todas tiene el mismo formato.\
-El dataset operaciones tiene la columna 'Grado' categórica.\
-El dataset aditivos no tiene columnas categóricas.\
-El dataset calidad tiene la columna 'Brillo_75' categórica.


In [354]:
#df_operaciones.head()
#df_operaciones.describe()
#df_operaciones.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 132481 entries, 0 to 132480
Data columns (total 14 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   Date             132481 non-null  object
 1   Corte_hoja       132481 non-null  object
 2   Grado            132481 non-null  object
 3   Velocidad_pope   132481 non-null  object
 4   Gramaje_cara     132481 non-null  object
 5   Gramaje_Reverso  132481 non-null  object
 6   Gramaje_Tripa    132481 non-null  object
 7   Gramaje_prote    132481 non-null  object
 8   Caudal_cara      132481 non-null  object
 9   Caudal_Prote     132481 non-null  object
 10  Caudal_Reverso   132481 non-null  object
 11  Caudal_Tripa     132481 non-null  object
 12  secado_1         132481 non-null  object
 13  secado_2         132481 non-null  object
dtypes: object(14)
memory usage: 14.2+ MB


In [484]:
#df_aditivos.head()
#df_aditivos.describe()
df_aditivos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 132481 entries, 0 to 132480
Data columns (total 19 columns):
 #   Column                    Non-Null Count   Dtype         
---  ------                    --------------   -----         
 0   Date                      132481 non-null  datetime64[ns]
 1   Resina_CARA               132447 non-null  float64       
 2   Resina_TRIPA              132447 non-null  float64       
 3   Resina_REVERSO            132447 non-null  float64       
 4   Agente_Retencion          132447 non-null  float64       
 5   Almidon_cationico_cara    132447 non-null  float64       
 6   Soda_CARA                 131328 non-null  float64       
 7   Soda_REVERSO              129475 non-null  float64       
 8   Agente_Drenaje            132447 non-null  float64       
 9   polymin_SK                132447 non-null  float64       
 10  Sulfato_CARA              132445 non-null  float64       
 11  Sulfato_TRIPA             132447 non-null  float64       
 12  Su

In [488]:
#df_calidad.head()
#df_calidad.describe()
df_calidad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 132481 entries, 0 to 132480
Data columns (total 18 columns):
 #   Column              Non-Null Count   Dtype         
---  ------              --------------   -----         
 0   Date                132481 non-null  datetime64[ns]
 1   Gramaje_pope        132480 non-null  float64       
 2   Humedad_pope        132480 non-null  float64       
 3   Espesor_pope        132480 non-null  float64       
 4   Rigidez_Long        131354 non-null  float64       
 5   Rigidez_Transv      131354 non-null  float64       
 6   Scott_Bond          131359 non-null  float64       
 7   Blancura_Cara       131354 non-null  float64       
 8   Blancura_Reverso    101585 non-null  float64       
 9   Cobb_Cara           131101 non-null  float64       
 10  Cobb_Cara_Base      129911 non-null  float64       
 11  Cobb_Tripa          131101 non-null  float64       
 12  Cobb_Reverso        101351 non-null  float64       
 13  Brillo_75           132481 no

In [358]:
# Convierte las columnas de fecha que vienen en formato cadena de caracteres

df_operaciones['Date'] = pd.to_datetime(df_operaciones['Date'], format='%m/%d/%Y %H:%M:%S')
df_aditivos['Date'] = pd.to_datetime(df_aditivos['Date'], format='%m/%d/%Y %H:%M:%S')
df_calidad['Date'] = pd.to_datetime(df_calidad['Date'], format='%m/%d/%Y %H:%M:%S')

In [359]:
print(df_operaciones.columns)

Index(['Date', 'Corte_hoja', 'Grado', 'Velocidad_pope', 'Gramaje_cara',
       'Gramaje_Reverso', 'Gramaje_Tripa', 'Gramaje_prote', 'Caudal_cara',
       'Caudal_Prote', 'Caudal_Reverso', 'Caudal_Tripa', 'secado_1',
       'secado_2'],
      dtype='object')


In [362]:
# Convierte las columnas del dataset operaciones con datos float que vienen en formato cadena de caracteres

columnas_float_op =['Corte_hoja', 'Velocidad_pope', 'Gramaje_cara','Gramaje_Reverso', 'Gramaje_Tripa',
                 'Gramaje_prote', 'Caudal_cara', 'Caudal_Prote', 'Caudal_Reverso', 'Caudal_Tripa', 'secado_1',
                 'secado_2']

for columna in columnas_float_op:
    df_operaciones[columna] = pd.to_numeric(df_operaciones[columna], errors='coerce')
    

In [364]:
print(df_aditivos.columns)

Index(['Date', 'Resina_CARA', 'Resina_TRIPA', 'Resina_REVERSO',
       'Agente_Retencion', 'Almidon_cationico_cara', 'Soda_CARA ',
       'Soda_REVERSO', 'Agente_Drenaje', 'polymin_SK', 'Sulfato_CARA',
       'Sulfato_TRIPA', 'Sulfato_REVERSO', 'Almidon_CARA-PROTECCION',
       'Almidon_PROTECCION-TRIPA', 'Almidon_REVERSO', 'Encolante_PRE',
       'Encolante_CARA', 'Encolante_REVERSO'],
      dtype='object')


In [366]:
# Convierte las columnas del dataset aditivos con datos float que vienen en formato cadena de caracteres

columnas_float_ad = ['Resina_CARA', 'Resina_TRIPA', 'Resina_REVERSO',
       'Agente_Retencion', 'Almidon_cationico_cara', 'Soda_CARA ',
       'Soda_REVERSO', 'Agente_Drenaje', 'polymin_SK', 'Sulfato_CARA',
       'Sulfato_TRIPA', 'Sulfato_REVERSO', 'Almidon_CARA-PROTECCION',
       'Almidon_PROTECCION-TRIPA', 'Almidon_REVERSO', 'Encolante_PRE',
       'Encolante_CARA', 'Encolante_REVERSO']

for columna in columnas_float_ad:
    df_aditivos[columna] = pd.to_numeric(df_aditivos[columna], errors='coerce')

In [368]:
print(df_calidad.columns)

Index(['Date', 'Gramaje_pope', 'Humedad_pope', 'Espesor_pope', 'Rigidez_Long',
       'Rigidez_Transv', 'Scott_Bond', 'Blancura_Cara', 'Blancura_Reverso',
       'Cobb_Cara', 'Cobb_Cara_Base', 'Cobb_Tripa', 'Cobb_Reverso',
       'Brillo_75', 'Lisura_PPS', 'Tersura_Sheffield', 'Blistering_Cara',
       'Blistering_Reverso'],
      dtype='object')


In [370]:
# Convierte las columnas del dataset calidad con datos float que vienen en formato cadena de caracteres

columnas_float_ca = ['Gramaje_pope', 'Humedad_pope', 'Espesor_pope', 'Rigidez_Long',
       'Rigidez_Transv', 'Scott_Bond', 'Blancura_Cara', 'Blancura_Reverso',
       'Cobb_Cara', 'Cobb_Cara_Base', 'Cobb_Tripa', 'Cobb_Reverso', 'Lisura_PPS', 'Tersura_Sheffield', 'Blistering_Cara',
       'Blistering_Reverso']

for columna in columnas_float_ca:
    df_calidad[columna] = pd.to_numeric(df_calidad[columna], errors='coerce')

Guardo estos nuevos dataset como archivos csv para usarlos en etapas posteriores del proyecto

In [373]:
df_operaciones.to_csv('ds_operaciones.csv', index=False)
df_aditivos.to_csv('ds_aditivos.csv', index=False)
df_calidad.to_csv('ds_calidad.csv', index=False)

#### Los tres dataset tienen el mismo número de muestras (132481 filas)

In [403]:
print('Tamaño del dataset')
print(f' Cantidad de muestras del dataset de operaciones: ', len(df_operaciones['Date']))
print(f' Cantidad de muestras del dataset de aditivos: ', len(df_aditivos['Date']))
print(f' Cantidad de muestras del dataset de calidad: ', len(df_calidad['Date']))

Tamaño del dataset
 Cantidad de muestras del dataset de operaciones:  132481
 Cantidad de muestras del dataset de aditivos:  132481
 Cantidad de muestras del dataset de calidad:  132481


#### Los tres dataset comienzan al mismo tiempo 2024-05-01 00:00

In [405]:
print('Fecha de inicio')
print(f' Fecha de inicio dataset operaciones: ', df_operaciones['Date'].min())
print(f' Fecha de inicio dataset operaciones: ', df_aditivos['Date'].min())
print(f' Fecha de inicio dataset operaciones: ', df_calidad['Date'].min())

Fecha de inicio
 Fecha de inicio dataset operaciones:  2024-05-01 00:00:00
 Fecha de inicio dataset operaciones:  2024-05-01 00:00:00
 Fecha de inicio dataset operaciones:  2024-05-01 00:00:00


#### Los tres dataset finalizan al mismo tiempo 2024-08-01 00:00

In [407]:
print('Fecha de termino')
print(f' Fecha de termino dataset operaciones: ', df_operaciones['Date'].max())
print(f' Fecha de termino dataset operaciones: ', df_aditivos['Date'].max())
print(f' Fecha de termino dataset operaciones: ', df_calidad['Date'].max())

Fecha de termino
 Fecha de termino dataset operaciones:  2024-08-01 00:00:00
 Fecha de termino dataset operaciones:  2024-08-01 00:00:00
 Fecha de termino dataset operaciones:  2024-08-01 00:00:00


In [411]:
print('Datos no numéricos del dataset de operaciones')
for i in columnas_float_op:
    print(f' {i}: ', df_operaciones[i].isna().sum())

Datos no numéricos del dataset de operaciones
 Corte_hoja:  34
 Velocidad_pope:  20
 Gramaje_cara:  208
 Gramaje_Reverso:  208
 Gramaje_Tripa:  208
 Gramaje_prote:  208
 Caudal_cara:  19
 Caudal_Prote:  19
 Caudal_Reverso:  19
 Caudal_Tripa:  19
 secado_1:  43
 secado_2:  43


In [413]:
print('Datos no numéricos del dataset de aditivos')
for i in columnas_float_ad:
    print(f' {i}: ', df_aditivos[i].isna().sum())

Datos no numéricos del dataset de aditivos
 Resina_CARA:  34
 Resina_TRIPA:  34
 Resina_REVERSO:  34
 Agente_Retencion:  34
 Almidon_cationico_cara:  34
 Soda_CARA :  1153
 Soda_REVERSO:  3006
 Agente_Drenaje:  34
 polymin_SK:  34
 Sulfato_CARA:  36
 Sulfato_TRIPA:  34
 Sulfato_REVERSO:  34
 Almidon_CARA-PROTECCION:  34
 Almidon_PROTECCION-TRIPA:  34
 Almidon_REVERSO:  34
 Encolante_PRE:  43
 Encolante_CARA:  34
 Encolante_REVERSO:  43


#### Las columnas 'Encolante_PRE' 'Encolante_CARA' 'Encolante_REVERSO' tiene solo ceros y NaN

In [482]:
print(f'Valores únicos en Encolante_PRE ', np.unique(df_aditivos['Encolante_PRE']))
print(f'Valores únicos en Encolante_CARA ', np.unique(df_aditivos['Encolante_CARA']))
print(f'Valores únicos en Encolante_PRE ', np.unique(df_aditivos['Encolante_REVERSO']))

Valores únicos en Encolante_PRE  [ 0. nan]
Valores únicos en Encolante_CARA  [ 0. nan]
Valores únicos en Encolante_PRE  [ 0. nan]


#### En el dataset de calidad hay tres columnas que tienen un gran número de valores NaN con más de 30000

In [440]:
print('Datos no numéricos del dataset de calidad')
for i in columnas_float_ca:
    print(f' {i}: ', df_calidad[i].isna().sum())

Datos no numéricos del dataset de calidad
 Gramaje_pope:  1
 Humedad_pope:  1
 Espesor_pope:  1
 Rigidez_Long:  1127
 Rigidez_Transv:  1127
 Scott_Bond:  1122
 Blancura_Cara:  1127
 Blancura_Reverso:  30896
 Cobb_Cara:  1380
 Cobb_Cara_Base:  2570
 Cobb_Tripa:  1380
 Cobb_Reverso:  31130
 Lisura_PPS:  1127
 Tersura_Sheffield:  1127
 Blistering_Cara:  1251
 Blistering_Reverso:  30979


#### La columna 'Brillo_75' solo tiene dos valores y no son valores de calidad, por lo que pueden ser errores de medición

In [491]:
print(f'Valores unicos en Brillo_75 ',np.unique(df_calidad['Brillo_75']))

Valores unicos en Brillo_75  ['I/O Timeout' 'Intf Shut']


#### La columna categórica 'Grado' que contiene los tipos de fabricación tiene dos tipos de datos que no corresponden

In [498]:
print(np.unique(df_operaciones['Grado']))

['2C205B' '2C215M' '2C225B' '2C235M' '2C240B' '2C245C' '2C250M' '2C295C'
 '2C325M' '2F225M' '3C250B' '3C255C' '3C260M' '3C265C' '3C270B' '3C275M'
 '3C285C' '3C290B' '3C295C' '3C295M' '3C310C' '3C310M' '3C311M' '3C315B'
 '3C325M' '3C330C' '3C340B' '3C345C' '3C370B' '3C370C' '3C370M'
 'Cargando...' 'I/O Timeout']
